### ISSUE TITLE
no colour in vim in a Docker container when using contour

### ISSUE THREAD

**Issue by Yaraslaut** [contour-terminal/contour#1244]
### Discussed in https://github.com/contour-terminal/contour/discussions/1240

<div type='discussions-op-text'>

<sup>Originally posted by **mufasachan** October  9, 2023</sup>
I edit files in running containers using Vim, primarily for prototyping a Dockerfile. I recently installed  contour, and it's fantastic. However, when I edit files in my container with Vim, the colors are missing. I suspect I may have overlooked something in the configurations (either Docker or contour).

---

When I execute the following commands:

```bash
docker run -it debian:stable-slim bash
>> apt update && apt install -y vim
>> vim
```

I don't see any colors when editing a file, for instance, `.bashrc`. The main screen also lacks colors. If I try to view highlight groups (`:hi`), the list of all highlight groups appears, but it's entirely in black and white.

Moreover, I've noticed that executing any command in the bash within the container works fine. For example:

``` bash
# In container
>> ls --color
>> echo -e "\e[31mRed Text\e[0m"
```

--- 

I've added this post to the discussion as I don't believe the issue is related to Contour. I suspect my lack of understanding regarding terminal emulators and Docker is the cause. If you can identify or replicate this behavior and investigate why it doesn't work, I would be greatly appreciative :grin: . 

I wonder why the container's bash seems to handle colors well, while Vim within the container fails to output colors.</div>

**Comment by uspasojevic96** [comment]
Check if you added contour terminfo to the docker image

**Comment by Yaraslaut** [comment]
This is true that terminfo is not inside docker image, its just raw image with installed vim

**Comment by uspasojevic96** [comment]
just set TERM to xterm-256color then, it should work

**Comment by mufasachan** [comment]
It works, Yaraslaut provided links in its answer in discussion and I set this variable to this value and it worked.

**Comment by Yaraslaut** [comment]
This is great.
@mufasachan can you provide docker setting that fixes this issue?

**Comment by mufasachan** [comment]
Oops, sorry, sure.

```bash
docker run -it -e TERM='screen-256color' debian:stable-slim  bash
>> apt update && apt install -y vim
>> vim # Colorful this time
```

**Comment by Yaraslaut** [comment]
Thanks, i am closing it then
